app/src/main/java/org/openimis/imispolicies/OverViewPolicies.java

```diff
@@ -531,101 +482,76 @@ public void LoginDialogBox() {
         alertDialog.show();
     }
 
-    private int getControlNumber(final JSONObject order, final String SmsRequired) throws IOException {
-        new Thread() {
-            public void run() {
-                runOnUiThread(() -> pd = ProgressDialog.show(OverViewPolicies.this, "", getResources().getString(R.string.Get_Control_Number)));
+    private int getControlNumber(final JSONObject order, final String SmsRequired) {
```

Comment:
I see similar code in OverViewControlNumbers class. In case the code is the same and to satisfy the DRY principle, we should have the method declared once and reused multiple times.